### PR TITLE
bugfix (@nestjs/microservices) pass replyTo to deserializer

### PR DESCRIPTION
**Pull request by johnbiundo**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

It's currently impossible to write a deserializer that reliably handles transforming the shape of an inbound (non-nest-conforming) NATS message to nest-conforming.

This makes it impossible to write fully interoperable NATS consumers in Nest.  For example, assume a distributed application system utilizes NATS messages to make "requests" (a NATS abstraction that publishes a message with a `replyTo` field that the message recipient uses to direct a published message (response) back to the requester).  Assume that in this system, a sample message is published on the topic `'get-customer'`, along with a standard NATS `replyTo` field, with the payload 
```json
{
  id: 2
}
```

If we would like to migrate the existing (non-Nest) service provider for the `'get-customer'` request to Nest, Nest must be able to accept these existing messages without requiring requestors (existing non-Nest NATS-client apps that publish these messages) to **change their request protocol**/message format.

In the current system, this is not possible to do with a deserializer as the deserializer has no way of knowing whether an inbound message is a **request** or an **event**.  Requests need to have `id` fields added to them, whereas events **must not** (else, they end up incorrectly triggering a response publication).

The motivation for this change is to eliminate this constraint.

Issue Number: https://github.com/nestjs/docs.nestjs.com/issues/406

## What is the new behavior?

Adding the `replyTo` field to the list of `options` available in a user-supplied `deserialize()` method implementation allows the `deserialize()` method to determine whether an incoming message from an external (non-nest) publisher is a [request-response](https://docs.nestjs.com/microservices/nats#request-response) style message or an [event](https://docs.nestjs.com/microservices/nats#event-based) style message.

This solves the problem described above.  For example, the following deserializer properly discriminates events from requests, and adds the required `id` field to requests:

```typescript
class OmniDeserializer implements ConsumerDeserializer {

  deserialize(value: any, options?: Record<string, any>) {

    if (this.isInternal(value)) {
      return value;
    } else {
      const packet = {
        pattern: undefined,
        data: value,
      };
    }

    if (options && !isUndefined(options.replyTo)) {
      const id = uuid();
      Object.assign(packet, { id });
    }
    
    return packet;
  }

  isInternal(value: any): boolean {
    if (
      !isUndefined((value as IncomingRequest).pattern) &&
      !isUndefined((value as IncomingRequest).data) &&
      !isUndefined((value as IncomingRequest).id)
    ) {
      return true;
    }
    return false;
  }
}
```

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
Oops... commit message references `serializer`, but the change is to the`deserialize` call.

### CODE REVIEW

**Comment by johnbiundo** [comment]
@kamilmysliwiec I've added a small commit that affects one line of `client-nats.ts`.  The purpose of this is to round out full support for NATS interoperability.  I had forgotten that I made this change locally and left it out of the prior commit.  

The purpose of this change is to **not** fail in `createSubscriptionHandler()` (not return `undefined`) if the incoming response does not have a message id.  That condition will **only happen when accepting a response message from an external source** (all Nest <-> Nest messages always have an id, and this check remains valid for those cases).  

Without this change, it is not possible to accept an incoming response message from an external source as there's no way to generate an ID that will match on the request and the response.  However, such an ID is **not needed** (nor possible to generate) in such an incoming external response.  All that is needed is the `replyTo` subject.

In short, this change remains fully backward compatible (including the perverse case where someone is actually **using** the ID field in an external message); and it **makes it possible** to send requests to an external NATS consumers from a Nest `ClientProxy`.

**Comment by kamilmysliwiec** [comment]
Thank you John!
